assets/src/components/calendar.js

```diff
@@ -182,7 +183,9 @@ function Calendar({
         internalAsyncEvents.push(asev);
       });
 
-      setEvents(internalAsyncEvents);
+      const eventObj = removeDuplicateEvents(internalAsyncEvents);
```

Comment:
I think this useEffect needs a lot of comments. I have no idea why this code is needed and what it is trying to accomplish. And why are there duplicated events?